### PR TITLE
types(vue.config.js): add descriptions for config

### PR DESCRIPTION
**Pull request by vegerot**
**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Docs
- [x] Underlying tools
- [x] Other, please describe: Config descriptions

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

**Other information:**
Personally, I have found it frustrating working with `vue.config.js`, due to lack of type information available.  Yesterday I discovered https://github.com/vuejs/vue-cli/issues/2138#issuecomment-484520356 , and was delighted to see I had a bit more info, but still not as much as I'd like.  

This commit adds JSDoc documentation (from `docs/config/README.md`) to add better editor support for `vue.config.js`

### CODE REVIEW

**Review by fangbinwei — CHANGES_REQUESTED** [review]
Thank you 😃, just a small problem

File: packages/@vue/cli-service/types/ProjectOptions.d.ts [hunk 1]
```diff
@@ -23,37 +23,121 @@ interface ExtractOptions {
...
   css?: CSSOptions;
 
+  /**
+   * A function that will receive an instance of `ChainableConfig` powered by [webpack-chain](https://github.com/mozilla-neutrino/webpack-chain)
+   */
   chainWebpack?: (config: ChainableWebpackConfig) => void;
+  /**
+   * Default: `[]`
+   * If you want to explicitly transpile a dependency with Babel, you can list it in this option
+   */
```
[the remark this answers is not in the document]
Comment (vegerot) [reply packages/@vue/cli-service/types/ProjectOptions.d.ts]: Fixed in 65a6148

File: packages/@vue/cli-service/types/ProjectOptions.d.ts [hunk 2]
```diff
@@ -23,37 +23,120 @@ interface ExtractOptions {
...
   integrity?: boolean;
 
   css?: CSSOptions;
 
+  /**
+   * A function that will receive an instance of `ChainableConfig` powered by [webpack-chain](https://github.com/mozilla-neutrino/webpack-chain)
+   */
   chainWebpack?: (config: ChainableWebpackConfig) => void;
+  /**
+   * Set webpack configuration.  If the value is `Object`, will be merged into config.  If value is `Function`, will receicve current config as argument
```
Comment (fangbinwei) [inline packages/@vue/cli-service/types/ProjectOptions.d.ts:123]:
```suggestion
   * Set webpack configuration.  If the value is `Object`, will be merged into config.  If value is `Function`, will receive current config as argument
```

**Review by fangbinwei — CHANGES_REQUESTED** [review]
Correct spelling

File: packages/@vue/cli-service/types/ProjectOptions.d.ts [hunk 3]
```diff
@@ -23,37 +23,120 @@ interface ExtractOptions {
...
+   * By default, only files that ends in `*.module.[ext]` are treated as CSS modules
+   */
   requireModuleExtension?: boolean;
+  /**
+   * Default: `true`
+   * Whether to extract CSS in your components into a standalone CSS files (instead of inlined in JavaScript and injected dynamically)
+   */
   extract?: boolean | ExtractOptions;
+  /**
+   * Default: `true`
```
Comment (fangbinwei) [inline packages/@vue/cli-service/types/ProjectOptions.d.ts:37]:
```suggestion
   * Default: `false`
```

Comment (fangbinwei) [reply packages/@vue/cli-service/types/ProjectOptions.d.ts:37]: sourceMap is disabled by default

**Review by fangbinwei — CHANGES_REQUESTED** [review]

**Review by fangbinwei — APPROVED** [review]
Thanks a lot!

File: packages/@vue/cli-service/types/ProjectOptions.d.ts [hunk 4]
```diff
@@ -23,37 +23,120 @@ interface ExtractOptions {
...
+  /**
+   * Default: `{}`
+   * Pass options to CSS-related loaders
+   */
   loaderOptions?: LoaderOptions;
 }
 
 interface ProjectOptions {
+  /**
+   * Default: `'/'`
```
Comment (fangbinwei) [inline packages/@vue/cli-service/types/ProjectOptions.d.ts:50]:
```suggestion
   * Default: `'/'`
   *
```

Comment (fangbinwei) [reply packages/@vue/cli-service/types/ProjectOptions.d.ts:50]: It seems adding a new line after all `Default`, the editor can give more friendly  parameter info.
image

If no new line, 
image

test by windows vscode

**Review by fangbinwei — CHANGES_REQUESTED** [review]

File: packages/@vue/cli-service/types/ProjectOptions.d.ts [hunk 5]
```diff
@@ -23,37 +23,120 @@ interface ExtractOptions {
...
+   * A function that will receive an instance of `ChainableConfig` powered by [webpack-chain](https://github.com/mozilla-neutrino/webpack-chain)
+   */
   chainWebpack?: (config: ChainableWebpackConfig) => void;
+  /**
+   * Set webpack configuration.  If the value is `Object`, will be merged into config.  If value is `Function`, will receive current config as argument
+   */
   configureWebpack?: WebpackOptions | ((config: WebpackOptions) => (WebpackOptions | void));
 
+  /**
+   * Default: `default`
```
Comment (fangbinwei) [inline packages/@vue/cli-service/types/ProjectOptions.d.ts:128]:
```suggestion
   * Default: `'default'`
```

Comment (fangbinwei) [reply packages/@vue/cli-service/types/ProjectOptions.d.ts:128]: single quotes for string

**Review by fangbinwei — CHANGES_REQUESTED** [review]

**Review by fangbinwei — APPROVED** [review]
Thanks a lot!

**Pull request merged**